### PR TITLE
Support Netberg Platform

### PR DESCRIPTION
**Pull request by raymondhuey**
Add Netberg Aurora 620 and Aurora 720 Platform Support

### CODE REVIEW

**Comment by sonoble** [comment]
Confirmed to install on 620 and 720 using modified machine.conf platform names.  Is there an ONIE patch for this?

**Comment by raymondhuey** [comment]
@sonoble Yes, there is an ONIE patch for Netberg's platforms and we are going to verify it and push it to the ONIE Github.

**Comment by raymondhuey** [comment]
@sonoble The ONIE patch for Netberg's platforms has been pushed to ONIE Github.

**Comment by raymondhuey** [comment]
Hi @sonoble,
The ONIE patch for Netberg's platforms had been merged to master.
Please review this pull request.
Thanks.

**Comment by sonoble** [comment]
I have built the new ONIE master, installed it, installed ONL and the system is functioning as I would expect.  Need to wait for @jnealtowns to look at the code before merging.

File: packages/base/any/kernels/3.16-lts/patches/driver-i2c-bus-intel-ismt-netberg-aurora.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
+diff -Nu a/drivers/i2c/busses/i2c-ismt.c b/drivers/i2c/busses/i2c-ismt.c
```
Comment (jnealtowns) [inline packages/base/any/kernels/3.16-lts/patches/driver-i2c-bus-intel-ismt-netberg-aurora.patch]: What is the purpose of this change?

File: packages/base/any/kernels/3.16-lts/patches/driver-igb-netberg-aurora.patch [hunk 2]
```diff
@@ -0,0 +1,599 @@
+diff -Nu a/drivers/net/ethernet/intel/igb/bcm_phy.c b/drivers/net/ethernet/intel/igb/bcm_phy.c
```
Comment (jnealtowns) [inline packages/base/any/kernels/3.16-lts/patches/driver-igb-netberg-aurora.patch]: These header definitions are already available in include/linux/brcmphy.h. 
54616 support is already available in e1000_82575 and e1000_hw. 

What is needed with this new file?

File: packages/base/any/kernels/3.16-lts/patches/driver-i2c-bus-intel-ismt-netberg-aurora.patch (hunk 1 above)
Comment (raymondhuey) [inline packages/base/any/kernels/3.16-lts/patches/driver-i2c-bus-intel-ismt-netberg-aurora.patch]: We found that some AOC locked the i2c-ismt bus at the beginning of inserting in machine and it causes i2c bus access timeout when reading EEPROM data from it, so we added "retry" here.

File: packages/base/any/kernels/3.16-lts/patches/driver-igb-netberg-aurora.patch (hunk 2 above)
Comment (raymondhuey) [inline packages/base/any/kernels/3.16-lts/patches/driver-igb-netberg-aurora.patch]: The function "bcm54616s_linkup" and "bcm54616_config_init" in "bcm_phy.c" is needed for Netberg platforms and we will remove unused and re-define definitions in this file.

**Comment by raymondhuey** [comment]
Hi @jnealtowns 
The patch file "driver-igb-netberg-aurora.patch" had been updated.
Please review it.
Thanks.

File: packages/base/any/kernels/3.16-lts/patches/driver-i2c-bus-intel-ismt-netberg-aurora.patch (hunk 1 above)
Comment (jnealtowns) [inline packages/base/any/kernels/3.16-lts/patches/driver-i2c-bus-intel-ismt-netberg-aurora.patch]: @raymondhuey Lets remove this change from your pull request for the moment since it would affect other platforms and get the rest of your code merged. Then we can work separately on the issue you are describing.

File: packages/platforms/netberg/vendor-config/src/python/netberg/__init__.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/python
+
+from onl.platform.base import *
+
+class OnlPlatformNetberg(OnlPlatformBase):
```
Comment (jnealtowns) [inline packages/platforms/netberg/vendor-config/src/python/netberg/__init__.py]: Please inherit from one of the existing PortConfig classes as well. 
See https://github.com/opencomputeproject/OpenNetworkLinux/blob/master/packages/platforms/accton/x86-64/x86-64-accton-as5712-54x/platform-config/r0/src/python/x86_64_accton_as5712_54x_r0/__init__.py#L4

File: packages/platforms/netberg/x86-64/x86-64-netberg-aurora-620-rangeley/onlp/builds/src/x86_64_netberg_aurora_620_rangeley/module/src/sysi.c [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+ * This function will be called if onlp_sysi_platform_get()
+ * returns a platform name that is not equal to the current platform.
+ *
+ * If you are compatible with the given platform then return ONLP_STATUS_OK.
+ * If you can are not compatible return ONLP_STATUS_E_UNSUPPORTED.
+ * - This is fatal and will abort platform initialization.
+ */
+
+int
+onlp_sysi_platform_set(const char* name)
```
Comment (jnealtowns) [inline packages/platforms/netberg/x86-64/x86-64-netberg-aurora-620-rangeley/onlp/builds/src/x86_64_netberg_aurora_620_rangeley/module/src/sysi.c]: You should not accept all platforms here. Please remove.

File: packages/platforms/netberg/x86-64/x86-64-netberg-aurora-620-rangeley/onlp/builds/src/x86_64_netberg_aurora_620_rangeley/module/src/sysi.c [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+{
+    int rv;
+    uint8_t buffer[512];
+    uint8_t data[256];
+    int len;
+    int i, j;
+
+    memset(buffer, 0, sizeof(buffer));
+    memset(data, 0, sizeof(data));
+    rv = onlp_file_read(buffer, sizeof(buffer), &len, SYS_HWMON2_PREFIX "/eeprom");
```
Comment (jnealtowns) [inline packages/platforms/netberg/x86-64/x86-64-netberg-aurora-620-rangeley/onlp/builds/src/x86_64_netberg_aurora_620_rangeley/module/src/sysi.c]: It looks like you have device file that produces a hexdump of the eeprom (which you then convert to a binary blob for decoding). Is there really no access to the binary data directly?

File: packages/platforms/netberg/x86-64/x86-64-netberg-aurora-720-rangeley/platform-config/r0/src/python/x86_64_netberg_aurora_720_rangeley_r0/__init__.py [hunk 6]
```diff
@@ -0,0 +1,17 @@
+from onl.platform.base import *
+from onl.platform.netberg import *
+
+class OnlPlatform_x86_64_netberg_aurora_720_rangeley_r0(OnlPlatformNetberg,
+                                                OnlPlatformPortConfig_32x100):
+    PLATFORM='x86-64-netberg-aurora-720-rangeley-r0'
+    MODEL="AURORA720"
+    SYS_OBJECT_ID=".8.1"
```
Comment (jnealtowns) [inline packages/platforms/netberg/x86-64/x86-64-netberg-aurora-720-rangeley/platform-config/r0/src/python/x86_64_netberg_aurora_720_rangeley_r0/__init__.py]: The 620 and the 720 cannot share the same SYS_OBJECT_ID

File: packages/platforms/netberg/x86-64/x86-64-netberg-aurora-620-rangeley/onlp/builds/src/x86_64_netberg_aurora_620_rangeley/module/src/sysi.c (hunk 4 above)
Comment (raymondhuey) [inline packages/platforms/netberg/x86-64/x86-64-netberg-aurora-620-rangeley/onlp/builds/src/x86_64_netberg_aurora_620_rangeley/module/src/sysi.c]: It will be removed in next commit.

File: packages/base/any/kernels/3.16-lts/patches/driver-i2c-bus-intel-ismt-netberg-aurora.patch (hunk 1 above)
Comment (raymondhuey) [inline packages/base/any/kernels/3.16-lts/patches/driver-i2c-bus-intel-ismt-netberg-aurora.patch]: It will be removed in the next commit.

File: packages/platforms/netberg/x86-64/x86-64-netberg-aurora-620-rangeley/onlp/builds/src/x86_64_netberg_aurora_620_rangeley/module/src/sysi.c (hunk 5 above)
Comment (raymondhuey) [inline packages/platforms/netberg/x86-64/x86-64-netberg-aurora-620-rangeley/onlp/builds/src/x86_64_netberg_aurora_620_rangeley/module/src/sysi.c]: It will be changed to access the binary data directly in the next commit.

File: packages/platforms/netberg/x86-64/x86-64-netberg-aurora-720-rangeley/platform-config/r0/src/python/x86_64_netberg_aurora_720_rangeley_r0/__init__.py (hunk 6 above)
Comment (raymondhuey) [inline packages/platforms/netberg/x86-64/x86-64-netberg-aurora-720-rangeley/platform-config/r0/src/python/x86_64_netberg_aurora_720_rangeley_r0/__init__.py]: It will be changed in the next commit.

File: packages/platforms/netberg/vendor-config/src/python/netberg/__init__.py (hunk 3 above)
Comment (raymondhuey) [inline packages/platforms/netberg/vendor-config/src/python/netberg/__init__.py]: I cannot understand it.
I had checked the file of accton, I cannot find any inherit information in it.
https://github.com/opencomputeproject/OpenNetworkLinux/blob/master/packages/platforms/accton/vendor-config/src/python/accton/__init__.py

**Comment by raymondhuey** [comment]
Hi @jnealtowns,
The code had been updated. Please review it.
Thanks.

**Comment by jnealtowns** [comment]
Thank you @raymondhuey !
